integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala

```diff
@@ -493,4 +495,49 @@ object PreAggregateUtil {
     updatedPlan
   }
 
+  /**
+   * This method will start load process on the data map
+   */
+  def startDataLoadForDataMap(parentCarbonTable: CarbonTable,
+      dataMapIdentifier: TableIdentifier,
+      queryString: String,
+      segmentToLoad: String,
+      validateSegments: Boolean,
+      sparkSession: SparkSession): Unit = {
+    CarbonSession.threadSet(
```

Comment:
Do not use thread local to pass parameters, pass them explicitly to the command

Reply:
This parameter is used to specify the segments to scan in CarbonScanRDD. Therefore i cannot pass it explicitly. As discussed with @gvramana and @ravipesala this was only option that was agreed upon.